# HG changeset 9522 patch
From: emellor@leeni.uk.xensource.com
# Node ID b3751c780aa5b8f511dffc4114102b4ca110a11b
# Parent  c5a833aec2abec5436038377aa2fb6be50dfafad
Subject: The attached patch fixes a race condition that occurs if after the
backend has been removed there are still requests pending. This patch
flushed all the pending requests.

Signed-off-by: Stefan Berger <stefanb@us.ibm.com>
Acked-By: Jan Beulich <jbeulich@novell.com>

Index: head-2006-04-05/drivers/xen/tpmback/xenbus.c
===================================================================
--- head-2006-04-05.orig/drivers/xen/tpmback/xenbus.c	2006-04-06 11:49:28.000000000 +0200
+++ head-2006-04-05/drivers/xen/tpmback/xenbus.c	2006-04-06 11:51:22.000000000 +0200
@@ -55,6 +55,7 @@ static int tpmback_remove(struct xenbus_
 		be->backend_watch.node = NULL;
 	}
 	if (be->tpmif) {
+		vtpm_release_packets(be->tpmif, 0);
 		tpmif_put(be->tpmif);
 		be->tpmif = NULL;
 	}
